### DISCUSSION TITLE
Checkbox "caption" feature

### DISCUSSION THREAD

**Discussion by groucho86** [zauberzeug/nicegui#1126] [Ideas / Feature Requests]
Would be great if checkbox's caption could be implemented:
https://quasar.dev/vue-components/checkbox#with-qitem

**Comment by rodja** [comment]
@groucho86 thanks for the suggestion. But Quasars checkboxes do not have a caption. The example you've linked is just a combination of multiple elements: 

![Screen Shot 2023-07-05 at 04 57 07](https://github.com/zauberzeug/nicegui/assets/131391/851896f4-316c-4682-bee8-32f5cb153e89)

```html
      <q-item tag="label" v-ripple>
        <q-item-section avatar>
          <q-checkbox v-model="color" val="orange" color="orange" />
        </q-item-section>
        <q-item-section>
          <q-item-label>Orange</q-item-label>
          <q-item-label caption>With description</q-item-label>
        </q-item-section>
      </q-item>
```

NiceGUI does not yet implement `ui.list` and `ui.item` (see #512). But the arrangement of a checkbox and two labels can be archived very easily:

```py
with ui.row():
    ui.checkbox().classes('mt-[-2px]')
    with ui.column().classes('gap-0'):
        ui.label('Orange')
        ui.label('With description').classes('text-xs text-gray-500')
```

In a `q-list` you would do it like this:

```py
with ui.element('q-list'):
    with ui.element('q-item'):
        with ui.element('q-item-section').props('avatar'):
            ui.checkbox()
        with ui.element('q-item-section'):
            ui.label('Orange')
            ui.label('With description').classes('text-xs text-gray-500')
```

**Reply by groucho86** [reply]
These little code snippets are incredibly useful to help me (us?) wrap my head around solutions. Thanks for sharing!
